zarr/storage.py

```diff
@@ -430,6 +436,28 @@ def init_array(
                          object_codec=object_codec,
                          dimension_separator=dimension_separator)
 
+    _init_array_attrs(store, path, attrs)
```

Comment:
What actually happens in terms of I/O operations when this is called. Specifically, in the case of V3 for example, do you know if the array metadata are re-read and re-written? That is my impression based on reading the code, but I find the `Attributes` class a bit hard to follow.

Reply:
I believe your impression is correct: calling `_init_array_attrs` calls [`Attributes._update_nosync`](https://github.com/d-v-b/zarr-python/blob/attrs_kwarg/zarr/attrs.py#L181), which eventually calls [`Attributes._get_nosync`](https://github.com/d-v-b/zarr-python/blob/attrs_kwarg/zarr/attrs.py#L40) and `Attributes._put_nosync`[https://github.com/d-v-b/zarr-python/blob/attrs_kwarg/zarr/attrs.py#L131], which read and write the entire attrs blob from storage respectively.

Reply:
It looks like I can use `Attributes.put` instead of `Attributes.update` to avoid a read. I will switch to this approach.

Reply:
@rabernat for v3 it looks like there's unnecessary IO [here](https://github.com/d-v-b/zarr-python/blob/attrs_kwarg/zarr/attrs.py#L158), where we check if the attrs key exists in the store. Since the routines I'm adding are called at initialization time, there's no need for this check. I'm still thinking about ways to avoid this for the specific case of initialization.

Reply:
For this PR, let's just focus on the API change. We can collaborate on a subsequent PR that doesn't touch the API but focuses on i/o optimization.